src/components/accessories/billFilterForm/BillFilterForm.tsx

```diff
@@ -0,0 +1,167 @@
+import { useFormik } from "formik";
+import get from "lodash.get";
+import has from "lodash.has";
+import React, { FC, useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { object, string } from "yup";
+import {
+  formatAllFieldValues,
+  getFromFields,
+} from "../../../libraries/formDataHandling/functions";
+import { IState } from "../../../types";
+import AutocompleteField from "../autocompleteField/AutocompleteField";
+import DateField from "../dateField/DateField";
+import { initialFields } from "./consts";
+import { searchPatient } from "../../../state/patients/actions";
+
+import "./styles.scss";
+import { BillFilterProps, TValues } from "./types";
+import { PatientDTO } from "../../../generated";
+
+const BillFilterForm: FC<BillFilterProps> = ({
+  onSubmit,
+  className,
+  fields,
+}) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const [searchPatientParams, setSearchPatientParams] = useState({} as TValues);
+
+  const handlePatientSearchChange = (event: any, value: string) => {
+    setSearchPatientParams({
+      id: !isNaN(+value) ? value : "",
+      firstName: isNaN(+value) ? value : "",
+      secondName: "",
+      birthDate: "",
+      address: "",
+    });
+    dispatch(searchPatient(searchPatientParams));
+  };
+
+  const validationSchema = object({
+    toDate: string().test({
+      name: "toDate",
+      message: t("bill.validatetodate"),
+      test: function (value) {
+        return +value >= +this.parent.fromDate;
+      },
+    }),
+  });
```

Comment:
I think that here we can exapand the validationSchema requiring all the fields as mandatory and add some validation on the date format values.

Reply:
Patient field is  not mandatory to filter bills,  filter dates have default values. And those defaults values help to avoid empty data when the user reaches the page.

Reply:
Ok I see for the patient.

But what happens if I delete the Date from field's value?

![image](https://user-images.githubusercontent.com/8697550/135481345-a9aa21f4-50a4-4e2c-9d43-5a99d0eaf6bc.png)

I think that both dates should be mandatory and validate as valid date